spec/models/application_form_spec.rb

```diff
@@ -70,4 +70,61 @@ def application_form_with_course_option_for_provider_with(level:)
       application_form
     end
   end
+
+  describe '#blank_application?' do
+    context 'when a candidate has not made any alterations to their applicaiton' do
+      it 'returns true' do
+        application_form = create(:application_form)
+        expect(application_form.blank_application?).to be_truthy
+      end
+    end
+
+    context 'when a candidate has added a course choice' do
+      it 'returns false' do
+        application_form = create(:application_form)
+        create(:application_choice, application_form: application_form)
+        expect(application_form.blank_application?).to be_falsey
+      end
+    end
+
+    context 'when a candidate has added a job' do
+      it 'returns false' do
+        application_form = create(:application_form)
+        create(:application_work_experience, application_form: application_form)
+        expect(application_form.blank_application?).to be_falsey
+      end
+    end
+
+    context 'when a candidate has added a volunteering experience' do
+      it 'returns false' do
+        application_form = create(:application_form)
+        create(:application_volunteering_experience, application_form: application_form)
+        expect(application_form.blank_application?).to be_falsey
+      end
+    end
+
+    context 'when a candidate has added a qualificaion' do
+      it 'returns false' do
+        application_form = create(:application_form)
+        create(:application_qualification, application_form: application_form)
+        expect(application_form.blank_application?).to be_falsey
+      end
+    end
+
+    context 'when a candidate has added a reference' do
+      it 'returns false' do
+        application_form = create(:application_form)
+        create(:reference, application_form: application_form)
+        expect(application_form.blank_application?).to be_falsey
+      end
+    end
+
+    context 'when a candidate has added a work history break' do
+      it 'returns false' do
+        application_form = create(:application_form)
+        create(:application_work_history_break, application_form: application_form)
+        expect(application_form.blank_application?).to be_falsey
+      end
+    end
+  end
```

Comment:
Do we still need these exhaustive tests? They're probably testing stuff that's already tested elsewhere.

Reply:
💯 pulled most of them out